src/creatures/players/vip/player_vip.cpp

```diff
@@ -0,0 +1,247 @@
+/**
+ * Canary - A free and open-source MMORPG server emulator
+ * Copyright (©) 2019-2024 OpenTibiaBR <opentibiabr@outlook.com>
+ * Repository: https://github.com/opentibiabr/canary
+ * License: https://github.com/opentibiabr/canary/blob/main/LICENSE
+ * Contributors: https://github.com/opentibiabr/canary/graphs/contributors
+ * Website: https://docs.opentibiabr.com/
+ */
+
+#include "pch.hpp"
+
+#include "creatures/players/vip/player_vip.hpp"
+
+#include "io/iologindata.hpp"
+
+#include "game/game.hpp"
+#include "creatures/players/player.hpp"
+
+const uint8_t PlayerVIP::firstID = 1;
+const uint8_t PlayerVIP::lastID = 8;
+
+PlayerVIP::PlayerVIP(Player &player) :
+	m_player(player) { }
+
+size_t PlayerVIP::getMaxEntries() const {
+	if (m_player.group->maxVipEntries != 0) {
```

Comment:
It might be interesting to check m_player.group, as it is a raw pointer and although it is not common, it may be nullptr in some scenario where the player has not been constructed properly, such as in the waiting list